File: core/commands/registry/irules/b64decode.py
# Enriched from F5 iRules reference documentation.
"""b64decode -- Returns a string that is base-64 decoded."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/b64decode.html"


@register
class B64decodeCommand(CommandDef):
    name = "b64decode"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="b64decode",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Returns a string that is base-64 decoded.",
                synopsis=("b64decode ANY_CHARS",),
                snippet="Returns a string that is base-64 decoded.",
                source=_SOURCE,
                examples=("when RULE_INIT {\n   set ::key [AES::key]\n}"),
                return_value="b64decode <string> Returns a string that is base-64 decoded",
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="b64decode ANY_CHARS",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(init_only=True),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.UNKNOWN,
                    reads=True,
                    connection_side=ConnectionSide.GLOBAL,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/b64decode.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `b64decode` iRules command.
use crate::prelude::*;
use tcl_dialect::model::SpecSurface;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "b64decode",
        surface: Some(SpecSurface::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Returns a string that is base-64 decoded.",
            synopsis: &["b64decode ANY_CHARS"],
            snippet: "Returns a string that is base-64 decoded.",
            source: "https://clouddocs.f5.com/api/irules/b64decode.html",
            examples: "when RULE_INIT {\n   set ::key [AES::key]\n}",
            return_value: "b64decode <string> Returns a string that is base-64 decoded",
        }),
        forms: &[FormSpec {
            synopsis: "b64decode ANY_CHARS",
            ..FormSpec::DEFAULT
        }],
        side_effects: &[SideEffect {
            reads: true,
            connection_side: ConnectionSide::Global,
            ..SideEffect::DEFAULT
        }],
        traits: Traits::IS_UNESCAPE,
        ..CommandSpec::DEFAULT
    }
}
